From feaecf765368f2c273bea3a9fa641ff0da7e6b26 Mon Sep 17 00:00:00 2001
From: Laurent Cozic <laurent@cozic.net>
Date: Fri, 11 Jun 2021 20:17:45 +0200
Subject: [PATCH] Desktop, Mobile: Filter out form elements from note body to
 prevent potential XSS (thanks to Dmytro Vdovychinskiy for the PoC)

---
 packages/renderer/htmlUtils.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/renderer/htmlUtils.ts b/packages/renderer/htmlUtils.ts
index 1262ae6fd4..9a27fb31d5 100644
--- a/packages/renderer/htmlUtils.ts
+++ b/packages/renderer/htmlUtils.ts
@@ -158,7 +158,7 @@ class HtmlUtils {
 		// "link" can be used to escape the parser and inject JavaScript.
 		// Adding "meta" too for the same reason as it shouldn't be used in
 		// notes anyway.
-		const disallowedTags = ['script', 'iframe', 'frameset', 'frame', 'object', 'base', 'embed', 'link', 'meta', 'noscript'];
+		const disallowedTags = ['script', 'iframe', 'frameset', 'frame', 'object', 'base', 'embed', 'link', 'meta', 'noscript', 'button', 'form', 'input', 'select', 'textarea', 'option', 'optgroup'];
 
 		const parser = new htmlparser2.Parser({
 
